src/general/DialogInputHelper.cs

```diff
@@ -0,0 +1,98 @@
+using System.Collections.Generic;
+using System.Linq;
+using Godot;
+
+/// <summary>
+///   Helper class for all dialog escapes to be handled here.
+/// </summary>
+public class DialogInputHelper : NodeWithInput
+{
+    private static bool isHelperLoaded;
+
+    /// <summary>
+    ///   If this is not the first helper in Tree, disable this helper.
+    /// </summary>
+    private bool isDuplicate;
+
+    private List<WindowDialog> dialogs = new List<WindowDialog>();
+
+    public override void _EnterTree()
+    {
+        isDuplicate = isHelperLoaded;
+        if (!isHelperLoaded)
+        {
+            isHelperLoaded = true;
+        }
+
+        base._EnterTree();
+    }
+
+    public override void _ExitTree()
+    {
+        if (!isDuplicate)
+        {
+            isHelperLoaded = false;
+        }
+
+        base._ExitTree();
+    }
+
+    /// <summary>
+    ///   When all the nodes are ready search for all dialog windows in the Tree.
+    /// </summary>
+    public override void _Ready()
+    {
+        if (!isDuplicate)
+        {
+            var nodes = new List<Node>();
+            nodes.AddRange(GetTree().Root.GetChildren().OfType<Node>());
+
+            while (nodes.Count != 0)
+            {
+                var currentNode = nodes[0];
+
+                nodes.RemoveAt(0);
+
+                if (currentNode is WindowDialog asDialog)
+                {
+                    // Skip ProcessPanel and CheatMenu and OrganellePopup
+                    if (asDialog.Name == "ProcessPanel"
+                        || asDialog.Name == "CheatMenu"
+                        || asDialog.Name == "OrganellePopup")
```

Comment:
I kinda don't like this code, feels dirty to me to walk the scene tree hoping to find some dialogs. And then especially those explicit name checks are very prone to breaking in the future.
I think a better design would be a new custom dialog base type that should be used everywhere (we can write in our style guide that plain Godot dialogs should not be used).

Reply:
I could add the universal custom dialog type (and perhaps some kind of exclusive dialog detection as well(?)) in https://github.com/Revolutionary-Games/Thrive/pull/2208 if possible or maybe in a separate dedicated branch instead. So this could just utilize that for the dialog escape handling.

Reply:
> I think a better design would be a new custom dialog base type that should be used everywhere (we can write in our style guide that plain Godot dialogs should not be used).

I've tried this, we can add a WindowDialogWithInput base class, but many classes such as ConfirmationDialog in Godot are directly based from WindowDialog, which is unmodifiable. If we want to add a universal dialog class we must also add all derived classes as well. That's a lot of work.

Reply:
It's less work than to edit each place where a dialog is used, much easier to make custom dialog types for every Godot class and use those.

Reply:
With unified custom dialog type we can also specify some kind of UI animation behaviour that comes with it, for example a fade or scale tweening for when a popup is shown or hidden.